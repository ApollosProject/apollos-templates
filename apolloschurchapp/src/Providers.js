import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers, NavigationService } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import {
  LiveProvider,
  ACCEPT_FOLLOW_REQUEST,
} from '@apollosproject/ui-connected';
import { checkOnboardingStatusAndNavigate } from '@apollosproject/ui-onboarding';
import { ONBOARDING_VERSION } from './ui/Onboarding';

import ClientProvider, { client } from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      // TODO deprecated prop
      navigate={NavigationService.navigate}
      handleExternalLink={(url) => {
        const path = url.split('app-link/')[1];
        const [route, location] = path.split('/');
        if (route === 'content')
          NavigationService.navigate('ContentSingle', { itemId: location });
        if (route === 'nav')
          NavigationService.navigate(
            // turns "home" into "Home"
            location[0].toUpperCase() + location.substring(1)
          );
      }}
      actionMap={{
        // accept a follow request when someone taps "accept" in a follow request push notification
        acceptFollowRequest: ({ requestPersonId }) =>
          client.mutate({
            mutation: ACCEPT_FOLLOW_REQUEST,
            variables: { personId: requestPersonId },
          }),
      }}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() =>
          checkOnboardingStatusAndNavigate({
            client,
            navigation: NavigationService,
            latestOnboardingVersion: ONBOARDING_VERSION,
          })
        }
      >
        <AnalyticsProvider>
          <LiveProvider>
            <Providers
              themeInput={customTheme}
              iconInput={customIcons}
              {...props}
            />
          </LiveProvider>
        </AnalyticsProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
