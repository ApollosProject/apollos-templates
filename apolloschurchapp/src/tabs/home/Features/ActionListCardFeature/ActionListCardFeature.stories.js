import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';

import { BackgroundView, CenteredView } from '@apollosproject/ui-kit';
import ActionListFeature from '.';

storiesOf('ActionListCardFeature', module)
  .addDecorator((story) => (
    <BackgroundView>
      {/* eslint-disable-next-line react-native/no-inline-styles */}
      <CenteredView style={{ alignItems: 'stretch' }}>{story()}</CenteredView>
    </BackgroundView>
  ))
  .add('default', () => {
    const actions = [
      {
        id: 'fakeId1',
        title: 'Boom',
        subtitle: 'What',
        parentChannel: {
          id: '',
          name: '',
        },
        image: {
          sources: [
            {
              uri: 'https://picsum.photos/200/200',
            },
          ],
        },
      },
      {
        id: 'fakeId2',
        title: 'Boom',
        subtitle: 'What',
        parentChannel: {
          id: '',
          name: '',
        },
        image: {
          sources: [
            {
              uri: 'https://picsum.photos/200',
            },
          ],
        },
      },
      {
        id: 'fakeId3',
        title: 'Boom',
        subtitle: 'What',
        parentChannel: {
          id: '',
          name: '',
        },
        image: {
          sources: [
            {
              uri: 'https://picsum.photos/200',
            },
          ],
        },
      },
      {
        id: 'fakeId4',
        title: 'Boom',
        subtitle: 'What',
        parentChannel: {
          id: '',
          name: '',
        },
        image: {
          sources: [
            {
              uri: 'https://picsum.photos/200',
            },
          ],
        },
      },
    ];
    return <ActionListFeature actions={actions} />;
  })
  .add('isLoading', () => {
    const loadingStateData = [
      {
        id: 'fakeId1',
        title: '',
        subtitle: '',
        parentChannel: {
          id: '',
          name: '',
        },
        image: {
          sources: [
            {
              uri: '',
            },
          ],
        },
      },
      {
        id: 'fakeId2',
        title: '',
        subtitle: '',
        parentChannel: {
          id: '',
          name: '',
        },
        image: {
          sources: [
            {
              uri: '',
            },
          ],
        },
      },
      {
        id: 'fakeId3',
        title: '',
        subtitle: '',
        parentChannel: {
          id: '',
          name: '',
        },
        image: {
          sources: [
            {
              uri: '',
            },
          ],
        },
      },
      {
        id: 'fakeId4',
        title: '',
        subtitle: '',
        parentChannel: {
          id: '',
          name: '',
        },
        image: {
          sources: [
            {
              uri: '',
            },
          ],
        },
      },
    ];

    return <ActionListFeature actions={loadingStateData} isLoading />;
  });
