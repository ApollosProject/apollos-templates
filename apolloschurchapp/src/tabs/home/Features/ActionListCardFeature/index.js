import React, { memo } from 'react';
import PropTypes from 'prop-types';

import { styled, ActionListCard, H3, H6 } from '@apollosproject/ui-kit';

const StyledH6 = styled(({ theme }) => ({
  color: theme.colors.text.tertiary,
}))(H6);

const ActionListCardFeature = memo(
  ({
    actions,
    id,
    isLoading,
    onPressActionListButton,
    onPressActionItem,
    subtitle,
    title,
  }) => (
    <ActionListCard
      isLoading={isLoading}
      key={id}
      header={
        <>
          {isLoading || title ? ( // we check for isloading here so that they are included in the loading state
            <StyledH6 numberOfLines={1}>{title}</StyledH6>
          ) : null}
          {isLoading || subtitle ? <H3>{subtitle}</H3> : null}
        </>
      }
      actions={actions}
      onPressActionItem={onPressActionItem}
      onPressActionListButton={isLoading || onPressActionListButton} // we check for isloading here so that the button is included in the loading state
    />
  )
);

ActionListCardFeature.displayName = 'Features';

ActionListCardFeature.propTypes = {
  // TODO: refactor ActionListCard to safely render without an actions array.
  actions: PropTypes.arrayOf(PropTypes.shape({})).isRequired, // at least for the time being this is required
  id: PropTypes.number,
  isLoading: PropTypes.bool,
  onPressActionListButton: PropTypes.func,
  onPressActionItem: PropTypes.func,
  subtitle: PropTypes.string,
  title: PropTypes.string,
};

export default ActionListCardFeature;
