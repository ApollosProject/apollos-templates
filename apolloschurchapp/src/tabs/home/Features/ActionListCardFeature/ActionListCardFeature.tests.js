import React from 'react';
import renderer from 'react-test-renderer';

import Providers from '../../../../Providers';

import ActionListCardFeature from '.';

const actions = [
  {
    id: 'fakeId1',
    title: 'Boom',
    subtitle: 'What',
    parentChannel: {
      id: '',
      name: '',
    },
    image: {
      sources: [
        {
          uri: 'https://picsum.photos/200/200',
        },
      ],
    },
  },
  {
    id: 'fakeId2',
    title: 'Boom',
    subtitle: 'What',
    parentChannel: {
      id: '',
      name: '',
    },
    image: {
      sources: [
        {
          uri: 'https://picsum.photos/200',
        },
      ],
    },
  },
  {
    id: 'fakeId3',
    title: 'Boom',
    subtitle: 'What',
    parentChannel: {
      id: '',
      name: '',
    },
    image: {
      sources: [
        {
          uri: 'https://picsum.photos/200',
        },
      ],
    },
  },
  {
    id: 'fakeId4',
    title: 'Boom',
    subtitle: 'What',
    parentChannel: {
      id: '',
      name: '',
    },
    image: {
      sources: [
        {
          uri: 'https://picsum.photos/200',
        },
      ],
    },
  },
];

describe('The Onboarding LandingScreen component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <ActionListCardFeature actions={actions} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with a title', () => {
    const tree = renderer.create(
      <Providers>
        <ActionListCardFeature actions={actions} title={'Boom'} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
