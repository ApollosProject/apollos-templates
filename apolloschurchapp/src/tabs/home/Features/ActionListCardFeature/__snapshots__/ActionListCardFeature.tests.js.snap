// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Onboarding LandingScreen component should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 16,
      "marginHorizontal": 16,
      "marginVertical": 12,
      "shadowColor": "rgba(165, 165, 165, 0.6)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 6,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 16,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderBottomWidth": 0,
          "borderColor": "rgba(165, 165, 165, 0.6)",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "marginBottom": 16,
        }
      }
    >
      <View
        accessible={true}
        clickable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingHorizontal": 16,
              "paddingVertical": 4,
            }
          }
        >
          <View
            animate={null}
            customAnimate={null}
            onReady={false}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#A5A5A5",
                "borderRadius": 16,
                "height": 64,
                "marginRight": 16,
                "width": 64,
              }
            }
          >
            <Image
              onLoad={[Function]}
              source={
                Array [
                  Object {
                    "cache": "force-cache",
                    "uri": "https://picsum.photos/200/200",
                  },
                ]
              }
              style={
                Object {
                  "backgroundColor": "#A5A5A5",
                  "borderRadius": 16,
                  "height": 64,
                  "marginRight": 16,
                  "opacity": 0,
                  "width": 64,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "borderBottomWidth": 0.5,
                "borderColor": "rgba(165, 165, 165, 0.6)",
                "flex": 1,
                "height": 68,
                "justifyContent": "center",
                "marginTop": 6.4,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "InterUI-Medium",
                  "fontSize": 14,
                  "lineHeight": 21,
                }
              }
            >
              Boom
            </Text>
            <Text
              bold={false}
              ellipsizeMode="tail"
              italic={false}
              numberOfLines={2}
              style={
                Object {
                  "color": "#A5A5A5",
                  "fontFamily": "InterUI-Regular",
                  "fontSize": 12,
                  "lineHeight": 18,
                }
              }
            >
              What
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        clickable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingHorizontal": 16,
              "paddingVertical": 4,
            }
          }
        >
          <View
            animate={null}
            customAnimate={null}
            onReady={false}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#A5A5A5",
                "borderRadius": 16,
                "height": 64,
                "marginRight": 16,
                "width": 64,
              }
            }
          >
            <Image
              onLoad={[Function]}
              source={
                Array [
                  Object {
                    "cache": "force-cache",
                    "uri": "https://picsum.photos/200",
                  },
                ]
              }
              style={
                Object {
                  "backgroundColor": "#A5A5A5",
                  "borderRadius": 16,
                  "height": 64,
                  "marginRight": 16,
                  "opacity": 0,
                  "width": 64,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "borderBottomWidth": 0.5,
                "borderColor": "rgba(165, 165, 165, 0.6)",
                "flex": 1,
                "height": 68,
                "justifyContent": "center",
                "marginTop": 6.4,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "InterUI-Medium",
                  "fontSize": 14,
                  "lineHeight": 21,
                }
              }
            >
              Boom
            </Text>
            <Text
              bold={false}
              ellipsizeMode="tail"
              italic={false}
              numberOfLines={2}
              style={
                Object {
                  "color": "#A5A5A5",
                  "fontFamily": "InterUI-Regular",
                  "fontSize": 12,
                  "lineHeight": 18,
                }
              }
            >
              What
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        clickable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingHorizontal": 16,
              "paddingVertical": 4,
            }
          }
        >
          <View
            animate={null}
            customAnimate={null}
            onReady={false}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#A5A5A5",
                "borderRadius": 16,
                "height": 64,
                "marginRight": 16,
                "width": 64,
              }
            }
          >
            <Image
              onLoad={[Function]}
              source={
                Array [
                  Object {
                    "cache": "force-cache",
                    "uri": "https://picsum.photos/200",
                  },
                ]
              }
              style={
                Object {
                  "backgroundColor": "#A5A5A5",
                  "borderRadius": 16,
                  "height": 64,
                  "marginRight": 16,
                  "opacity": 0,
                  "width": 64,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "borderBottomWidth": 0.5,
                "borderColor": "rgba(165, 165, 165, 0.6)",
                "flex": 1,
                "height": 68,
                "justifyContent": "center",
                "marginTop": 6.4,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "InterUI-Medium",
                  "fontSize": 14,
                  "lineHeight": 21,
                }
              }
            >
              Boom
            </Text>
            <Text
              bold={false}
              ellipsizeMode="tail"
              italic={false}
              numberOfLines={2}
              style={
                Object {
                  "color": "#A5A5A5",
                  "fontFamily": "InterUI-Regular",
                  "fontSize": 12,
                  "lineHeight": 18,
                }
              }
            >
              What
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        clickable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingHorizontal": 16,
              "paddingVertical": 4,
            }
          }
        >
          <View
            animate={null}
            customAnimate={null}
            onReady={false}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#A5A5A5",
                "borderRadius": 16,
                "height": 64,
                "marginRight": 16,
                "width": 64,
              }
            }
          >
            <Image
              onLoad={[Function]}
              source={
                Array [
                  Object {
                    "cache": "force-cache",
                    "uri": "https://picsum.photos/200",
                  },
                ]
              }
              style={
                Object {
                  "backgroundColor": "#A5A5A5",
                  "borderRadius": 16,
                  "height": 64,
                  "marginRight": 16,
                  "opacity": 0,
                  "width": 64,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "borderBottomWidth": 0.5,
                "borderColor": "rgba(165, 165, 165, 0.6)",
                "flex": 1,
                "height": 68,
                "justifyContent": "center",
                "marginTop": 6.4,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "InterUI-Medium",
                  "fontSize": 14,
                  "lineHeight": 21,
                }
              }
            >
              Boom
            </Text>
            <Text
              bold={false}
              ellipsizeMode="tail"
              italic={false}
              numberOfLines={2}
              style={
                Object {
                  "color": "#A5A5A5",
                  "fontFamily": "InterUI-Regular",
                  "fontSize": 12,
                  "lineHeight": 18,
                }
              }
            >
              What
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
